### PR TITLE
Upgrade to rgbds 0.4.0, and do miscellaneous fixes

### PR DESCRIPTION
**Pull request by Rangi42**
- fail for rgbds < 0.4.0
- `rst <Label>`
- `ds <count>, <byte>`
- `shift <N>`
- ASCII "\r"
- Sorted .sym file

### CODE REVIEW

**Comment by Rangi42** [comment]
This does not yet resolve the `DAYCARE`/`DAY_CARE` inconsistency from #700 because I'm not sure which to go with. `DAYCARE` is more common, but in-game it's spelled "DAY-CARE".

This does not yet use `LOAD`/`ENDL` from rgbds 0.4.0 for `WriteOAMDMACodeToHRAM.PushOAM` and `hTransferVirtualOAM` because that fails to build the `jr` correctly.

**Comment by Rangi42** [comment]
Re: #700, I think the `DAY_CARE`/`DAYCARE` constants are already fine. Most use "`DAY_CARE`" to match in-game "DAY-CARE": there's `ENGINE_DAY_CARE_*`, `EVENT_DAY_CARE_*`, the `DAY_CARE` map, `SPRITE_DAY_CARE_MON_1/2`, `ROUTE34_DAY_CARE_MON_1/2`, etc.

There are three kinds of constant that use "`DAYCARE`": `DAYCAREMAN/DAYCARELADY_*_F`, `DAYCARETEXT_*`, and the `object_event` constants `DAYCARE_GRAMPS/GRANNY`. All of those are following the pattern where spaces get taken out of names for brevity. For example, `NEWBARKTOWN_SILVER`, `BATTLETOWERTEXT_INTRO`, or `LUCKYNUMBERSHOW_GAME_OVER_F`.

Also this doesn't need to wait for `LOAD`/`ENDL` to work properly, since `WriteOAMDMACodeToHRAM.PushOAM` is fine without them (the hard-coded `hTransferVirtualOAM:: ds 10` size is alright for now). So this should be ready to merge.

**Comment by mid-kid** [comment]
Except for the `*TEXT` constants, I think object event constants and event flags constants should match their respective map's names.

**Comment by Rangi42** [comment]
`EVENT_DAY_CARE_*` do match `map_const DAY_CARE`. Do you mean that all the `object_const_def`s should be updated? Such as `AZALEAMART_COOLTRAINER_M` → `AZALEA_MART_COOLTRAINER_M`?

**Comment by mid-kid** [comment]
In all honesty - yes. It's kind of arbitrary to make a difference with something that's related to a very exact map. But if it's a lot, it can be left for a different PR.

**Comment by Rangi42** [comment]
If we get file-local defines (https://github.com/rednex/rgbds/issues/342), I'd like them to be even shorter: no need for the map name in every constant, with or without underscores, since they're only ever used locally in the map script.

**Comment by mid-kid** [comment]
Agreed, it's something I wanted to consider as a result of #631, not only with defines, but also with the labels. The only two that need to be prefixed are the two that are exported, `_MapScripts` and `_MapEvents`.

File: FAQ.md [hunk 1]
```diff
@@ -38,11 +39,11 @@ You need to install `gcc`. If you're using Cygwin, re-run its setup, and at "Sel
 
 ### "ERROR: `UNION` already defined"
 
-Download [**rgbds 0.3.9**][rgbds] or newer. Older versions will not work.
+Download [**rgbds 0.4.0**][rgbds] or newer. Older versions will not work.
 
 ### "Expression must be 8-bit"
```
Comment (mid-kid) [inline FAQ.md:44]: Add one with `ERROR: Macro not defined`. It turns up with the ascii thing.

File: INSTALL.md [hunk 2]
```diff
@@ -157,7 +157,7 @@ If `rgbds` is not available, you'll also need these:
 To install **rgbds**:
 
 ```bash
-git clone -b v0.3.9 --depth=1 https://github.com/rednex/rgbds
+git clone -b v0.4.0 --depth=1 https://github.com/rednex/rgbds
 sudo make -C rgbds CFLAGS=-O2 install
```
Comment (mid-kid) [inline INSTALL.md:161]: Remove `CFLAGS` from each of these. `-O3` is the default now.

File: constants.asm [hunk 3]
```diff
@@ -1,3 +1,7 @@
+if __RGBDS_MINOR__ < 4
```
Comment (mid-kid) [inline constants.asm:1]:
```suggestion
if __RGBDS_MAJOR__ <= 0 && __RGBDS_MINOR__ < 4
```

**Comment by Rangi42** [comment]
Shall I change this to just `-Weverything` and merge?

**Comment by mid-kid** [comment]
Sure, we can change it at any point. Be sure to include it in RGBASMFLAGS though.

**Pull request merged**